src/core_plugins/input_control_vis/public/lineage/lineage_map.js

```diff
@@ -0,0 +1,25 @@
+import _ from 'lodash';
+
+export function getLineageMap(controlParamsList) {
+  function getControlParamsById(controlId) {
+    return controlParamsList.find((controlParams) => {
+      return controlParams.id === controlId;
+    });
+  }
+
+  const lineageMap = new Map();
+  controlParamsList.forEach((rootControlParams) => {
+    const lineage = [rootControlParams.id];
+    const getLineage = (controlParams) => {
+      if (_.has(controlParams, 'parent') && controlParams.parent !== '' && !lineage.includes(controlParams.parent)) {
+        lineage.push(controlParams.parent);
```

Comment:
If we stored the actual reference to the parent, could we skip the `getControl` in the VisController?

Reply:
No. There are two sets of state.
1) control parameters - these are stored on the visualization saved object
2) control object - this just lives in heap and contains the current display state

getLineageMap only deals with control parameters